### PR TITLE
Fixes AI being able to control not working APCs

### PR DESCRIPTION
**Pull request by Danchi299**
## What Does This PR Do
Currently Ctrl Clicking APCs as AI does not have any conditional checks, meaning it will toggle busted and AI control disabled APCs
This pull request adds few conditional checks before turning APC on.

This is the re-opened version of https://github.com/ParadiseSS13/Paradise/pull/22511

## Why It's Good For The Game
After AHelping in game I was told that this is a bug and it should be reported, so I fixed it instead.

## Testing
Joined as Crew, broke APC, cut AI wire in APC, Emagged APC.
Joined as AI, broken APC did not toggle, cut APC gave a warning message, Emagged APC worked the same.

## Changelog
:cl:
fix: Fixed AI being able to change breaker state of not working APCs.
fix: Fixed AI being able to change breaker state of APCs with cut AI wire.
/:cl:

### CODE REVIEW

**Comment by Danchi299** [comment]
I messed up on other pull request by not making branch before committing
had to make new pull request

**Review by JimKil3 — APPROVED** [review]
lgtm

**Review by Henri215 — APPROVED** [review]
repeating the approval here

File: code/_onclick/ai_onclick.dm [hunk 1]
```diff
@@ -177,6 +177,11 @@
 // APC
 
 /obj/machinery/power/apc/AICtrlClick(mob/living/user) // turns off/on APCs.
+	if(stat & BROKEN)
+		return
```
Comment (Burzah) [inline code/_onclick/ai_onclick.dm:181]:
```suggestion
		return
		
```
It is good practice to break up and space different logic.

**Review by Burzah — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**